A2017/tensorflow.ipynb

```diff
@@ -0,0 +1,892 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# TensorFlow tutorial\n",
+    "_MILA, November 2017_\n",
+    "\n",
+    "## Protip: browsing through the TensorFlow API\n",
+    "\n",
+    "The [devdocs.io](http://devdocs.io/) website is an amazing resource to browse through the TensorFlow Python API (as well as many other APIs such as numpy or the Python API itself).\n",
+    "\n",
+    "# Using TensorFlow at MILA\n",
+    "\n",
+    "The most straightforward way to access TensorFlow using the MILA software stack is through the `TODO: WRITEME` conda environment. To activate the `TODO: WRITEME` environment, use the command\n",
+    "\n",
+    "```bash\n",
+    "source activate TODO: WRITEME\n",
+    "```\n",
+    "\n",
+    "To return back to normal, simply use the bash command\n",
+    "\n",
+    "```bash\n",
+    "source deactivate\n",
+    "```\n",
+    "\n",
+    "# Importing TensorFlow\n",
+    "\n",
+    "TensorFlow is imported as a Python package using the following statement:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import tensorflow as tf"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "_Note: the `tensorflow` package is usually aliased to `tf` for convenience._"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Paradigm\n",
+    "\n",
+    "TensorFlow separates the _definition_ of computation from its _execution_.\n",
+    "\n",
+    "Computation is defined via a [_dataflow_](https://en.wikipedia.org/wiki/Dataflow_programming) _graph_, i.e., a graph where nodes represent units of computation and the edges represent the data consumed or produced by the computation.\n",
```

Comment:
About this, I think tensorflow graph is more like Theano graph, ie. a graph with 2 types of nodes: operation nodes and input/output nodes. Tensors are the input/output nodes. Edges just connect an input/output node to an operation node.

Reply:
This is the way the [TensorFlow documentation](https://www.tensorflow.org/programmers_guide/graphs) describes it.

Reply:
Ok, I see!